heartbeat/monitors/active/dialchain/dialchain.go

```diff
@@ -47,141 +35,34 @@ func (c *DialerChain) Clone() *DialerChain {
-	})
-}
 
-func (c *DialerChain) Build() (d transport.Dialer, err error) {
-	d = c.Net.Dialer
 	for _, layer := range c.Layers {
-		if d, err = layer.Builder(d); err != nil {
+		if d, err = layer.build(event, d); err != nil {
 			return nil, err
 		}
 	}
 	return
 }
 
-func (c *DialerChain) TestBuild() error {
-	_, err := c.Build()
-	return err
-}
-
-func (c *DialerChain) DialWithMeasurements(network, host string) (fields common.MapStr, conn net.Conn, err error) {
-	var dialer transport.Dialer
-	fields = common.MapStr{}
-	if dialer, err = c.BuildWithMeasures(fields); err == nil {
-		conn, err = dialer.Dial(network, host)
-	}
-	return
-}
-
-func (c *DialerChain) Dial(network, host string) (conn net.Conn, err error) {
-	var dialer transport.Dialer
-	if dialer, err = c.Build(); err == nil {
-		return dialer.Dial(network, host)
-	}
-	return
-}
-
 func (c *DialerChain) AddLayer(l Layer) {
 	c.Layers = append(c.Layers, l)
 }
 
-func measureEventRTT(event common.MapStr, name string) DialLayerCallback {
-	return &MeasureLayerRTTCB{Callback: func(start, end time.Time) {
-		event[name] = look.RTT(end.Sub(start))
-	}}
-}
-
-func LayerCBDialer(cb DialLayerCallback, d transport.Dialer) transport.Dialer {
-	return transport.DialerFunc(func(network, address string) (net.Conn, error) {
-		cb.Start()
-		c, err := d.Dial(network, address)
-		cb.Done(err)
-		return c, err
-	})
-}
-
-func LayerDeltaCBDialer(
-	cb DialLayerCallback,
-	dialer transport.Dialer,
-	layer func(transport.Dialer) (transport.Dialer, error),
-) (transport.DialerFunc, error) {
-	starter := transport.DialerFunc(func(network, address string) (net.Conn, error) {
-		c, err := dialer.Dial(network, address)
-		cb.Start()
-		return c, err
-	})
-
-	layerInstance, err := layer(starter)
-	if err != nil {
-		return nil, err
-	}
-
-	return func(network, address string) (net.Conn, error) {
-		c, err := layerInstance.Dial(network, address)
-		cb.Done(err)
-		return c, err
-	}, nil
-}
-
-func ConstAddrDialer(name, addr string, to time.Duration) NetDialer {
-	return NetDialer{name, transport.DialerFunc(func(network, _ string) (net.Conn, error) {
-		switch network {
-		case "tcp", "tcp4", "tcp6", "udp", "udp4", "udp6":
-		default:
-			return nil, fmt.Errorf("unsupported network type %v", network)
-		}
-
-		dialer := &net.Dialer{Timeout: to}
-		return dialer.Dial(network, addr)
-	})}
-}
-
-func ConstAddrLayer(addr string, l Layer) Layer {
-	return Layer{l.Name, func(d transport.Dialer) (transport.Dialer, error) {
-		forward, err := l.Builder(d)
-		if err != nil {
-			return nil, err
-		}
-
-		return transport.DialerFunc(func(network, _ string) (net.Conn, error) {
-			return forward.Dial(network, addr)
-		}), nil
-	}}
-}
-
-func TCPDialer(name string, to time.Duration) NetDialer {
-	return NetDialer{name, transport.NetDialer(to)}
-}
-
-func UDPDialer(name string, to time.Duration) NetDialer {
-	return NetDialer{name, transport.NetDialer(to)}
+func (c *DialerChain) TestBuild() error {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method DialerChain.TestBuild should have comment or be unexported